pkg/delete/internal/automation/delete.go

```diff
@@ -33,55 +33,56 @@ import (
 )
 
-type Client interface {
+type client interface {
 	Delete(ctx context.Context, resourceType automationAPI.ResourceType, id string) (automation.Response, error)
 	List(ctx context.Context, resourceType automationAPI.ResourceType) (automation.ListResponse, error)
 }
 
-func Delete(ctx context.Context, c Client, automationResource config.AutomationResource, entries []pointer.DeletePointer) error {
-
+func Delete(ctx context.Context, c client, automationResource config.AutomationResource, entries []pointer.DeletePointer) error {
 	logger := log.WithCtxFields(ctx).WithFields(field.Type(string(automationResource)))
 	logger.Info("Deleting %d config(s) of type %q...", len(entries), automationResource)
 
 	deleteErrs := 0
-
 	for _, e := range entries {
+		deleteErrs += deleteSingle(ctx, c, e)
+	}
 
-		logger := logger.WithFields(field.Coordinate(e.AsCoordinate()))
+	if deleteErrs > 0 {
+		return fmt.Errorf("failed to delete %d Automation objects(s) of type %q", deleteErrs, automationResource)
+	}
+	return nil
+}
 
-		id := e.OriginObjectId
-		if id == "" {
-			id = idutils.GenerateUUIDFromCoordinate(e.AsCoordinate())
-		}
+func deleteSingle(ctx context.Context, c client, dp pointer.DeletePointer) int {
+	logger := log.WithCtxFields(ctx).WithFields(field.Type(dp.Type), field.Coordinate(dp.AsCoordinate()))
 
-		logger.Debug("Deleting %v with id %q.", automationResource, id)
+	id := dp.OriginObjectId
+	if id == "" {
+		id = idutils.GenerateUUIDFromCoordinate(dp.AsCoordinate())
+	}
 
-		resourceType, err := automationutils.ClientResourceTypeFromConfigType(automationResource)
-		if err != nil {
-			logger.WithFields(field.Error(err)).Error("Failed to delete %v with ID %q: %v", automationResource, id, err)
-			deleteErrs++
-		}
+	logger.Debug("Deleting %v with id %q.", dp.Type, id)
 
-		_, err = c.Delete(ctx, resourceType, id)
-		if err != nil {
-			var apiErr api.APIError
-			if errors.As(err, &apiErr) {
-				if apiErr.StatusCode != http.StatusNotFound {
-					logger.WithFields(field.Error(err)).Error("Failed to delete %v with ID %q - rejected by API: %v", automationResource, id, err)
-					deleteErrs++
-				}
-			} else {
-				logger.WithFields(field.Error(err)).Error("Failed to delete %v with ID %q - network error: %v", automationResource, id, err)
-				deleteErrs++
+	resourceType, err := automationutils.ClientResourceTypeFromConfigType(config.AutomationResource(dp.Type))
+	if err != nil {
+		logger.WithFields(field.Error(err)).Error("Failed to delete %v with ID %q: %v", dp.Type, id, err)
+		return 1
+	}
+	_, err = c.Delete(ctx, resourceType, id)
+	if err != nil {
+		var apiErr api.APIError
+		if errors.As(err, &apiErr) {
+			if apiErr.StatusCode != http.StatusNotFound {
+				logger.WithFields(field.Error(err)).Error("Failed to delete %v with ID %q - rejected by API: %v", dp.Type, id, err)
+				return 1
 			}
+		} else {
+			logger.WithFields(field.Error(err)).Error("Failed to delete %v with ID %q - network error: %v", dp.Type, id, err)
```

Comment:
Is this really always a "network error"?

Reply:
We end here if an error (different from API.APIError) happens. Technically, it can be some bug in the core library, but mostly, that is some problem in network communication.